### ISSUE TITLE
Update to 2.02 breaks instanciation of epsg 31467

### ISSUE THREAD

**Issue by nylocx** [pyproj4/pyproj#198]
After updating to the latest version this simple line:
`Proj(init='epsg:31467')`
does not work more and errors our with

> CRSError: Invalid projection: +init=epsg:31467 +type=crs

As a workaround I used this:
`Proj('PROJCS["DHDN / Gauss-Kruger zone 3",GEOGCS["DHDN",DATUM["Deutsches_Hauptdreiecksnetz",SPHEROID["Bessel 1841",6377397.155,299.1528128,AUTHORITY["EPSG","7004"]],AUTHORITY["EPSG","6314"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.01745329251994328,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4314"]],UNIT["metre",1,AUTHORITY["EPSG","9001"]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",9],PARAMETER["scale_factor",1],PARAMETER["false_easting",3500000],PARAMETER["false_northing",0],AUTHORITY["EPSG","31467"],AXIS["Y",EAST],AXIS["X",NORTH]]')`

**Comment by snowman2** [comment]
The `+init` is deprecated in PROJ, but still enabled in pyproj. I recommend this approach for `EPSG` codes moving forward:
```
>>> from pyproj import Proj
>>> Proj("epsg:31467")
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)
>>> Proj(31467)
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)
```

**Comment by snowman2** [comment]
But, this works for me in 2.0.2
```
>>> from pyproj import Proj
>>> Proj(init="epsg:31467")
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)
```

**Comment by snowman2** [comment]
Can you provide information about the python version you are using, your operating system, and your method of installation?

**Comment by nylocx** [comment]
This is in my office but from memory this is a Conda installed python 3.6 and pyproj is installed via pip as a dependency of geopandas I think. This is on a Linux Server running jupyter Notebook in a docker container. I will check with the other syntax on Monday.

**Comment by snowman2** [comment]
Interesting, have you tried to install pyproj with conda? 
`conda install -c conda-forge pyproj`

If installed via conda, your environment needs to be activated to set the PROJ_LIB directory (depending on version - newest version does not need to be reactivated anymore I hear).

**Comment by nylocx** [comment]
I couldn't wait so I connected via an SSH tunnel. Here is the output for the following code:
```
print(Proj("epsg:31467"))
print(Proj(31467))
print(Proj(init='epsg:31467'))
````
```
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)
---------------------------------------------------------------------------
CRSError                                  Traceback (most recent call last)
<ipython-input-50-7dc33ff42676> in <module>
      1 print(Proj("epsg:31467"))
      2 print(Proj(31467))
----> 3 print(Proj(init='epsg:31467'))

/opt/conda/lib/python3.6/site-packages/pyproj/__init__.py in __init__(self, projparams, preserve_units, **kwargs)
    320         '116.366 39.867'
    321         """
--> 322         self.crs = CRS.from_user_input(projparams if projparams is not None else kwargs)
    323         # make sure units are meters if preserve_units is False.
    324         if not preserve_units and self.crs.is_projected:

/opt/conda/lib/python3.6/site-packages/pyproj/crs.py in from_user_input(cls, value)
    222             return cls.from_epsg(value)
    223         elif isinstance(value, dict):
--> 224             return cls(**value)
    225         elif isinstance(value, string_types):
    226             return cls.from_string(value)

/opt/conda/lib/python3.6/site-packages/pyproj/crs.py in __init__(self, projparams, **kwargs)
    144         # on case-insensitive filesystems).
    145         projstring = projstring.replace("+init=EPSG", "+init=epsg").strip()
--> 146         super(CRS, self).__init__(projstring)
    147 
    148     @classmethod

pyproj/_crs.pyx in pyproj._crs._CRS.__init__()

CRSError: Invalid projection: +init=epsg:31467 +type=crs
```

So the first two work fine the init one does not.
`pyproj.__version__`
> 2.0.2

Conda shows the same behavior.

**Comment by snowman2** [comment]
What is the value of PROJ_LIB?

**Comment by nylocx** [comment]
`!echo $PROJ_LIB`

> /opt/conda/lib/python3.6/site-packages/fiona/proj_data

**Comment by snowman2** [comment]
That's the wrong path for sure.

**Comment by snowman2** [comment]
What happens when you do `from pyproj.datadir import get_data_dir; print (get_data_dir())`?

**Comment by nylocx** [comment]
`!ls -la /opt/conda/lib/python3.6/site-packages/fiona/proj_data/`

> total 1679
> drwxr-sr-x 2 jovyan users     14 Mar 14 10:27 .
> drwxr-sr-x 7 jovyan users     31 Mar 14 10:27 ..
> -rw-r--r-- 1 jovyan users   1175 Mar 14 10:27 CH
> -rw-r--r-- 1 jovyan users 854691 Mar 14 10:27 epsg
> -rw-r--r-- 1 jovyan users 453436 Mar 14 10:27 esri
> -rw-r--r-- 1 jovyan users  76652 Mar 14 10:27 esri.extra
> -rw-r--r-- 1 jovyan users    694 Mar 14 10:27 GL27
> -rw-r--r-- 1 jovyan users  83441 Mar 14 10:27 IGNF
> -rw-r--r-- 1 jovyan users  19501 Mar 14 10:27 nad27
> -rw-r--r-- 1 jovyan users  16559 Mar 14 10:27 nad83
> -rw-r--r-- 1 jovyan users   6385 Mar 14 10:27 nad.lst
> -rw-r--r-- 1 jovyan users   3916 Mar 14 10:27 other.extra
> -rw-r--r-- 1 jovyan users    261 Mar 14 10:27 proj_def.dat
> -rw-r--r-- 1 jovyan users   7043 Mar 14 10:27 world

**Comment by nylocx** [comment]
```
from pyproj.datadir import get_data_dir
print (get_data_dir())
```

> /opt/conda/lib/python3.6/site-packages/pyproj/proj_dir/share/proj

**Comment by snowman2** [comment]
Can you try creating a new environment using pyproj=2.1.0?

**Comment by snowman2** [comment]
I am thinking that the two versions of the PROJ_LIB are conflicting here.

**Comment by nylocx** [comment]
It might be a bit difficult in my setup:
I have a Linux server running Docker and in that docker is running a JupyterHub that spawn user notebooks (in docker containers). I have to provide all users with all necessary packages in the docker image that is being spawned. So creating different environments will not be easy, I guess I have to figure out what packages are conflicting.

**Comment by nylocx** [comment]
I think you might close this issue as it seems to be related to my setup, but for reference and maybe others with the same problem I leave my complete `pip freeze` output.
```
absl-py==0.7.0
affine==2.2.2
alembic==1.0.8
appdirs==1.4.3
asn1crypto==0.24.0
astor==0.7.1
async-generator==1.10
attrs==19.1.0
autopep8==1.4.3
backcall==0.1.0
beautifulsoup4==4.7.1
black==18.9b0
blackcellmagic==0.0.1
bleach==3.1.0
bokeh==1.0.4
certifi==2019.3.9
cffi==1.12.2
chardet==3.0.4
Click==7.0
click-plugins==1.0.4
cligj==0.5.0
cloudpickle==0.8.0
conda==4.6.8
cryptography==2.6.1
cycler==0.10.0
Cython==0.29.6
cytoolz==0.9.0.1
dask==1.1.4
decorator==4.3.2
defusedxml==0.5.0
dill==0.2.9
distributed==1.26.0
EasyProcess==0.2.3
entrypoints==0.3
fastcache==1.0.2
Fiona==1.8.4
future==0.17.1
gast==0.2.2
geographiclib==1.49
geopandas==0.4.1
geopy==1.18.1
gmpy2==2.0.8
grpcio==1.19.0
gym==0.12.0
h5py==2.9.0
heapdict==1.0.0
idna==2.8
imageio==2.5.0
ipydatawidgets==4.0.0
ipykernel==5.1.0
ipyscales==0.3.0
ipython==7.3.0
ipython-genutils==0.2.0
ipyvolume==0.5.1
ipywebrtc==0.4.3
ipywidgets==7.4.2
isort==4.3.15
jedi==0.13.3
Jinja2==2.10
JSAnimation==0.1
jsonschema==3.0.1
jupyter-client==5.2.4
jupyter-contrib-core==0.3.3
jupyter-contrib-nbextensions==0.5.1
jupyter-core==4.4.0
jupyter-highlight-selected-word==0.2.0
jupyter-latex-envs==1.4.6
jupyter-nbextensions-configurator==0.4.1
jupyterhub==0.9.4
jupyterlab==0.35.4
jupyterlab-server==0.2.0
Keras==2.2.4
Keras-Applications==1.0.7
Keras-Preprocessing==1.0.9
kiwisolver==1.0.1
llvmlite==0.27.1
locket==0.2.0
lxml==4.3.2
Mako==1.0.7
Markdown==3.0.1
MarkupSafe==1.1.1
matplotlib==3.0.3
minio==4.0.12
mistune==0.8.4
mock==2.0.0
mpmath==1.1.0
msgpack==0.6.1
munch==2.3.2
nbconvert==5.4.1
nbformat==4.4.0
networkx==2.2
notebook==5.7.5
numba==0.42.1
numexpr==2.6.9
numpy==1.15.4
olefile==0.46
open3d-python==0.5.0.0
packaging==19.0
pamela==1.0.0
pandas==0.24.2
pandocfilters==1.4.2
parso==0.3.4
partd==0.3.9
patsy==0.5.1
pbr==5.1.3
pexpect==4.6.0
pickleshare==0.7.5
Pillow==5.4.1
prometheus-client==0.6.0
prompt-toolkit==2.0.9
protobuf==3.7.0
psutil==5.6.1
ptyprocess==0.6.0
pycodestyle==2.5.0
pycosat==0.6.3
pycparser==2.19
pycurl==7.43.0.2
pyglet==1.3.2
Pygments==2.3.1
pygpu==0.7.6
PyOpenGL==3.1.3b2
pyOpenSSL==19.0.0
pyparsing==2.3.1
pyproj==2.0.2
pyqualicision==1.0.0.dev3
pyrsistent==0.14.11
PySocks==1.6.8
python-dateutil==2.8.0
python-editor==1.0.4
python-oauth2==1.0.1
pythreejs==2.0.2
pytz==2018.9
PyVirtualDisplay==0.2.1
PyWavelets==1.0.2
PyYAML==5.1
pyzmq==18.0.1
rasterio==1.0.21
requests==2.21.0
ruamel-yaml==0.15.71
scikit-image==0.14.2
scikit-learn==0.20.3
scipy==1.2.1
seaborn==0.9.0
Send2Trash==1.5.0
Shapely==1.6.4.post2
six==1.12.0
snuggs==1.4.3
sortedcontainers==2.1.0
soupsieve==1.8
SQLAlchemy==1.3.1
statsmodels==0.9.0
sympy==1.3
tblib==1.3.2
tensorboard==1.12.2
tensorflow==1.12.0
tensorflow-estimator==1.13.0
termcolor==1.1.0
terminado==0.8.1
testpath==0.4.2
Theano==1.0.4
toml==0.10.0
toolz==0.9.0
tornado==6.0.1
traitlets==4.3.2
traittypes==0.2.1
urllib3==1.24.1
vincent==0.4.4
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.14.1
widgetsnbextension==3.4.2
xlrd==1.2.0
xvfbwrapper==0.2.9
zict==0.1.4
```

**Comment by snowman2** [comment]
This is useful, thanks! Are you installing everything using pip? Or do you use conda as well? Also, how is PROJ_LIB being set?

**Comment by nylocx** [comment]
I install some of the basic stuff using conda, but all of the geo an Machine learning libraries are installed via pip. I do not explicitly set the PROJ_LIB Variable I guess it was set by installing geopandas.

**Comment by snowman2** [comment]
Not sure how your PROJ_LIB is being set.

I did:
```
(base)$ conda create -n proj198
(base)$ conda activate proj198
(proj198) $ pip install geopandas==0.4.1 pyproj==2.1.0
(proj198)$ conda deactivate
(base)$ conda activate proj198
```
I checked for `PROJ_LIB` and didn't find it (`$ env | grep PROJ`). I wonder if it is being set somewhere in your container environment variables?

Then:
```python
>>> import pyproj
>>> pyproj.Proj(init="epsg:31467")
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)
>>> pyproj.__version__
'2.1.0'
```

**Comment by snowman2** [comment]
The problem is reproducible:
```
$ export PROJ_LIB=/home/snowal/miniconda/envs/proj198/lib/python3.6/site-packages/fiona/proj_data
(proj198) snowal@snowal-lx2:~$ python
Python 3.6.7 | packaged by conda-forge | (default, Feb 28 2019, 09:07:38) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyproj
>>> pyproj.Proj(init="epsg:31467")
proj_create: unknown datum potsdam
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/snowal/miniconda/envs/proj198/lib/python3.6/site-packages/pyproj/proj.py", line 295, in __init__
    self.crs = CRS.from_user_input(projparams if projparams is not None else kwargs)
  File "/home/snowal/miniconda/envs/proj198/lib/python3.6/site-packages/pyproj/crs.py", line 224, in from_user_input
    return cls(**value)
  File "/home/snowal/miniconda/envs/proj198/lib/python3.6/site-packages/pyproj/crs.py", line 146, in __init__
    super(CRS, self).__init__(projstring)
  File "pyproj/_crs.pyx", line 322, in pyproj._crs._CRS.__init__
pyproj.exceptions.CRSError: Invalid projection: +init=epsg:31467 +type=crs
```

**Comment by snowman2** [comment]
This will be addressed in the PROJ 6.1.0 release.

**Comment by snowman2** [comment]
Closing as this has been confirmed to be fixed in the 6.1.0 release:
```
$ export PROJ_LIB=$HOME/miniconda/envs/proj198/lib/python3.6/site-packages/fiona/proj_data
```
```
>>> import pyproj
>>> pyproj.Proj(init="epsg:31467")
Proj('+proj=tmerc +lat_0=0 +lon_0=9 +k=1 +x_0=3500000 +y_0=0 +ellps=bessel +units=m +no_defs', preserve_units=True)

```
